substrate/frame/src/lib.rs

```diff
@@ -334,6 +331,24 @@ pub mod testing_prelude {
 
 	/// Commonly used runtime traits for testing.
 	pub use sp_runtime::{traits::BadOrigin, StateVersion};
+
+	#[cfg(any(feature = "try-runtime", test))]
```

Comment:
The feature test here is only enabled when testing the crate `polkadot-sdk-frame`. I am not sure it is the intended behavior.

I think in general having `testing_prelude` feature gated with `std` is good enough to signify the testing environment.

(In general though I think a `testing` feature can also make sense.)
```suggestion
	#[cfg(any(feature = "try-runtime"))]
```